policy-controller/core/src/outbound.rs

```diff
@@ -38,18 +39,18 @@ pub struct OutboundPolicy {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub struct HttpRoute {
+pub struct OutboundRoute {
```

Comment:
This design allows GrpcRoutes to hold HttpRouteMatches and vice versa, which we then need to filter out later during the conversion to protobuf.  Just splitballing, but what do you think about parameterizing OutboundRoute on the match type?  Something like...

```
pub struct OutboundRoute<M> {
  ...
  pub rules: Vec<OutboundRouteRule<M>,
}
pub struct OutboundRouteRule<M> {
  pub matches: Vec<M>,
  ...
}
```

Which would allow us to have `OutboundRoute<HttpRouteMatch>` and `OutboundRoute<GrpcRouteMatch>` and ensure we don't have mismatched match types.

Reply:
This would also allow `convert_outbound_grpc_route` to operate on `OutboundRoute<GrpcRouteMatch>` specifically, and gives us some additional type safety.